core/src/smartcontracts/wasm.rs

```diff
@@ -543,11 +584,18 @@ pub mod state {
                 self.0.wsv
             }
         }
+
+        impl QueryHandle for Migrate<'_> {
```

Comment:
is it necessary to support lazy queries in Migrate?

Reply:
Why not? Users may want to perform queries from `migrate()` the same way they might want to do it in basic smart contracts

Reply:
By the way, this code is outdated, but `migrate()` can still execute queries